### DISCUSSION TITLE
DMN-02004 : HistoryTimeToLiveParser requires a non null config to be created

### DISCUSSION THREAD

**Discussion by intorenganzo** [camunda-community-hub/community#145] [Q&A]
I am having a couple of issues while upgrading a project.

Background story
I am currently using the 7.21.0-alpha2 with the upgrade of spring boot 3.2.1 with is not working with 7.20 version. Due to some spring framework class that was changed and was being by camunda (can't recall what the breaking change was)

The issue

Execution of `processEngineConfiguration.getDmnEngine().parseDecision(decisionName, decisionInputStream)` throws above `DMN-02004 : HistoryTimeToLiveParser requires a non null config to be created`. This snippet is executed within an asynchronous flow.

In bpm processes there are delegates that execute asynchronously the above snippet to evaluate the DMN results on multiple objects.

However the `org.camunda.bpm.engine.impl.context.Context.ProcessEngineConfiguration` inside `HistoryTimeToLiveParser.create()` (executed by `DecisionDefinitionHandler.validateAndSetHTTL`) is evaluating to null hence this error. This is not an issue in previous versions of Camunda and looks like it is related to enforcing historyTimeToLive in latest Camunda versions.

The workaround I have is to get the processEngineConfiguration and set it before calling `parseDecision` like in below
```
ProcessEngineConfigurationImpl pcImpl = processEngineConfiguration.getProcessEngine().getProcessEngineConfiguration();
Context.setProcessEngineConfiguration(pcImpl);

processEngineConfiguration.getDmnEngine().parseDecision(decisionName, is);
```

Couple of questions : 

- Is there a way to set `DecisionDefinitionHandler.skipEnforceTtl` through a property. It maybe preferable to opt-out for now and opt-in at a later time)
- I tried setting `camunda.bpm.generic-properties.properties.historyTimeToLive=P3D` in properties file so that I don't have to add to each BPMN and DMN but that did not work and had to add it in each file. I am missing something?

**Comment by xomiamoore** [comment]
Hey @leandremucyo, this looks like a question for the[ Camunda forum.](https://forum.camunda.io/) This GitHub Discussions page is just for extensions found in the Camunda Community Hub. Thanks!
